tests/components/evohome/conftest.py

```diff
@@ -153,3 +158,32 @@ async def setup_evohome(
         assert mock_client.account_info is not None
 
         yield mock_client
+
+
+def get_dhw_entity(hass: HomeAssistant) -> EvoDHW:
+    """Return the WaterHeater entity of the evohome integration."""
+
+    broker: EvoBroker = hass.data[DOMAIN]["broker"]
+
+    if (dhw := broker.tcs.hotwater) is None:
+        pytest.fail("DHW expected, but was not found")
+
+    entity_registry = er.async_get(hass)
+    entity_id = entity_registry.async_get_entity_id(
+        Platform.WATER_HEATER, DOMAIN, dhw._id
+    )
+
+    component: EntityComponent = hass.data.get(Platform.WATER_HEATER)  # type: ignore[assignment]
+    return next(e for e in component.entities if e.entity_id == entity_id)  # type: ignore[return-value]
```

Comment:
We should not touch the entity anymore after it is created. This currently is fetching the entity from the paltform and looking at the tests we use it to execute function calls. Instead of doing that directly to the entity, we should mutate the entities like we do in real life, with service calls.

We should avoid touching internals like `hass.data` as well